e2e-test/fluxd.yaml

```diff
@@ -0,0 +1,166 @@
+kind: Secret
+metadata:
+  creationTimestamp: null
+  name: flux-git-deploy
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  labels:
+    name: flux
+  name: flux
+  namespace: default
+---
+apiVersion: rbac.authorization.k8s.io/v1beta1
+kind: ClusterRole
+metadata:
+  labels:
+    name: flux
+  name: flux
+rules:
+  - apiGroups: ["*"]
+    resources: ["*"]
+    verbs: ["*"]
+  - nonResourceURLs: ["*"]
+    verbs: ["*"]
+---
+apiVersion: rbac.authorization.k8s.io/v1beta1
+kind: ClusterRoleBinding
+metadata:
+  labels:
+    name: flux
+  name: flux
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: flux
+subjects:
+  - kind: ServiceAccount
+    name: flux
+    namespace: default
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: flux-recv
+data:
+  fluxrecv.yaml: |
+    fluxRecvVersion: 1
+    endpoints: []
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: flux
+  namespace: default
+  labels:
+    name: flux
+    type: deployment
+spec:
+  replicas: 1
+  strategy:
+    type: RollingUpdate
+    rollingUpdate:
+      maxUnavailable: 0
+      maxSurge: 1
+  selector:
+    matchLabels:
+      name: flux
+  template:
+    metadata:
+      labels:
+        name: flux
+        type: deployment
+    spec:
+      serviceAccountName: flux
+      volumes:
+        - name: git-key
+          secret:
+            secretName: flux-git-deploy
+            defaultMode: 0400 # when mounted read-only, we won't be able to chmodª
```

Comment:
This is not mounted on any container?

Reply:
🤔

Reply:
It's a magic secret convention: https://github.com/fluxcd/flux/blob/a3319b09e362740963c7e971aa12ddb36387a24b/docs/guides/provide-own-ssh-key.md

I'm making it less hidden by using `--k8s-secret-name` flag

Reply:
Nice find 👍